domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1928,7 +1927,13 @@ QueryHelper buildQueryHelper(
           replyHandler.handle(Future.failedFuture(query.cause()));
           return;
         }
-        replyHandler.handle(Future.succeededFuture(resultSetMapper.apply(new TotaledResults(query.result(), total))));
+        ResultSet result = query.result();
+        Integer actualTotal = total;
+        if (queryHelper.limit > 0 && result.getNumRows() < queryHelper.limit) {
+          actualTotal = queryHelper.offset +result.getNumRows();
+        }
```

Comment:
This algorithm is a copy of the algorithm in PgUtil.
Please extract it into a method for reuse and write unit tests for this extracted method to show that it is correct.